### PR TITLE
add selector to abuser stats + add session data in search response

### PR DESCRIPTION
**Pull request by the-eater**
This adds the following to cluegetter:
- Abuse reports based on sasl_username instead of sender_domain (since sasl_username has more importance to us)
- Session data to the response of a search request.
- A key to mysql for sasl_username since it may now be used for abuse reports

### CODE REVIEW

File: src/cluegetter/http.go [hunk 1]
```diff
@@ -442,12 +466,12 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	rows, err := Rdbms.Query(`
-		SELECT sender_domain, count(*) amount
```
Comment (Freeaqingme) [inline src/cluegetter/http.go:445]: Even when selector == sasl_username, wouldn't you want to know the sender_domain as well? I'd suggest selecting them both.

File: assets/htmlTemplates/abusers.html [hunk 2]
```diff
@@ -23,6 +23,11 @@
                 {{ end }}
             </select>
         </div>
+        <div class="form-group">
+            <label for="selector">Selector</label>
+            <br>
+            <select id="" name="selector"><option value="sender_domain">Sender domain</option><option value="sasl_username">Sasl username</option></select>
```
Comment (Freeaqingme) [inline assets/htmlTemplates/abusers.html:29]: Why specify an empty id attribute?

File: assets/htmlTemplates/abusers.html [hunk 3]
```diff
@@ -23,6 +23,11 @@
                 {{ end }}
             </select>
         </div>
+        <div class="form-group">
+            <label for="selector">Selector</label>
+            <br>
```
Comment (Freeaqingme) [inline assets/htmlTemplates/abusers.html:28]: - &lt;br />

File: assets/htmlTemplates/abusers.html (hunk 2 above)
Comment (Freeaqingme) [inline assets/htmlTemplates/abusers.html:29]: Do you have a specific aversion to new lines?

File: src/cluegetter/http.go [hunk 4]
```diff
@@ -161,6 +161,8 @@ func httpHandlerMessageSearchEmail(w http.ResponseWriter, r *http.Request) {
 
 	rows, err := Rdbms.Query(`
 	SELECT m.id, m.date, CONCAT(m.sender_local, '@', m.sender_domain) sender, m.rcpt_count, m.verdict,
+           s.ip, s.reverse_dns, s.helo, s.sasl_username, s.sasl_method, s.cert_issuer, s.cert_subject,
+           s.cipher_bits, s.cipher, s.tls_version,
```
Comment (Freeaqingme) [inline src/cluegetter/http.go:165]: What use case do you have in mind for selecting all these fields? I'm trying to think in what kind of scenario the tls version that was used could be useful. The same applies to some of the other fields.

File: assets/htmlTemplates/abusers.html (hunk 2 above)
Comment (the-eater) [inline assets/htmlTemplates/abusers.html:29]: oh I generate html with emmet, which doesnt like new lines and gives an empty id, will fix

**Comment by Freeaqingme** [comment]
Did you run `go fmt`?

**Comment by the-eater** [comment]
Yes, I ran `make fmt` (its a thing?) as well as `go fmt ./src/cluegetter/*.go`

File: assets/htmlTemplates/abusers.html (hunk 2 above)
Comment (Freeaqingme) [inline assets/htmlTemplates/abusers.html:29]: No matter what you write your stuff with, I urge you to manually review your own diffs. Saves time for the both of us.

File: mysql.sql [hunk 5]
```diff
@@ -46,6 +46,7 @@ CREATE TABLE session (
   PRIMARY KEY (id),
   KEY cluegetter_instance (cluegetter_instance),
   KEY session_date (cluegetter_instance,date_connect),
+  KEY sasl_username (sasl_username),
```
Comment (Freeaqingme) [inline mysql.sql:49]: Please also add a DDL change to the DDL-changes file in the root.

**Pull request closed without merging**